update: deprecating android target (#4536)
JetBrains/kotlin-web-site: docs/topics/multiplatform/multiplatform-compatibility-guide.md

Android team from Google.

To open the way for the new solution from Google, we're renaming the `android` block to `androidTarget` in the current
Kotlin DSL. This is a temporary change that is necessary to free the short `android` name for the upcoming DSL
from Google.

**What's the best practice now?**

Rename all the occurrences of the `android` block to `androidTarget`. When the new plugin for the Android target support
is available, migrate to the DSL from Google. It will be the preferred option to work with Android in Kotlin Multiplatform
projects.

**When do the changes take effect?**

Here's the planned deprecation cycle:

> * 1.9.0: a deprecation warning is introduced when the `android` name is used in Kotlin Multiplatform projects
> * 2.1.0: report an error in such cases, causing the build to fail

Reviewer: This aligns better with the style of deprecation cycle lists:

```suggestion
* 1.9.0: introduce a deprecation warning when the `android` name is used in Kotlin Multiplatform projects
* 2.1.0: raise this warning to an error
```
